packages/firestore/src/core/transaction.ts

```diff
@@ -42,33 +44,12 @@ export class Transaction {
    */
   private lastWriteError: FirestoreError;
 
-  constructor(private datastore: Datastore) {}
-
-  private recordVersion(doc: MaybeDocument): void {
-    let docVersion: SnapshotVersion;
-
-    if (doc instanceof Document) {
-      docVersion = doc.version;
-    } else if (doc instanceof NoDocument) {
-      // For deleted docs, we must use baseVersion 0 when we overwrite them.
-      docVersion = SnapshotVersion.forDeletedDoc();
-    } else {
-      throw fail('Document in a transaction was a ' + doc.constructor.name);
-    }
+  /**
+   * Set of documents that have been written in the transaction.
```

Comment:
Optional: consider helping your readers understand why this is useful to track. For example, adding a sentence like "When there's more than one write to the same key in a transaction, any writes after the first are handled differently." could help.

Then again, could be overkill.